.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout commit
+      uses: actions/checkout@v1
+
+    - name: Checkout submodules
+      run: git submodule update --init --recursive
+
+    - name: Cache Cargo
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo
+        key: ${{ runner.os }}-cargo
+    
+    - name: Cache binaryen
+      uses: actions/cache@v1
+      with:
+        path: ~/binaryen
+        key: ${{ runner.os }}-binaryen
+    
+    - name: Cache target
+      uses: actions/cache@v1
+      with:
+        path: livesplit-core/target
+        key: ${{ runner.os }}-target
+
+    - name: Install Node
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+
+    - name: Install Rust
+      uses: hecrj/setup-rust-action@v1
+      with:
+        rust-version: 'nightly'
+
+    - name: Install target
+      run: rustup target add wasm32-unknown-unknown
+
+    - name: Install dependencies
+      run: |
+        set -e
+        sh .github/workflows/install.sh
+        source ~/.cargo/env || true
+
+    - name: Build application
+      run: bash .github/workflows/build.sh
+
+    - name: Deploy application
+      if: github.repository == 'LiveSplit/LiveSplitOne' && github.ref == 'refs/heads/master'
+      run: |
+        git config --global user.email "action@github.com"
+        git config --global user.name "GitHub Action"
+        git checkout -b gh-pages
+        mv dist/* .
+        git add :/
+        git commit -m "GitHub Pages build"
+        git push --force https://action:${{ secrets.LIVESPLIT_ONE_DEPLOY }}@github.com/LiveSplit/LiveSplitOne.git
```

Comment:
Is there no prebuilt Action for pushing?

Reply:
There is, but I had trouble getting it to work when I was setting up the build to push the `livesplit-core` docs. I'll try it again though.